### ISSUE TITLE
Efficiency refactor - run adjacentSquares only once for each square

### ISSUE THREAD

**Issue by LeanderMcL** [LeanderMcL/sunset-weaver#8]
As I'm matching, emptying and refilling squares, I'm running adjacentSquares over and over and over again for each square I touch. This is not very efficient. New plan: run adjacentSquares once for each square on grid creation, storing the resulting object as an attribute to each square, then accessing the attribute each time I need it later.
